### PR TITLE
Refactor the GLPI boot logic

### PR DESCRIPTION
**Pull request by Pierstoval**
With the proposed changes, the GLPI boot logic is now entirely handled in the `Kernel::boot()` sequence, it permits to use the same logic for all the different contexts (web request, cron script, console, tests bootstrap script, ...).

The database status checks logic have been moved into a dedicated request listener.

Also, some specific checks have been moved in the `front/cron.php` script as this is the only official CLI script that is not using the console yet and that therefore require this specific logic.

### CODE REVIEW

**Comment by cedric-anne** [comment]
As discussed yesterday, I put this PR back to draft, waiting for a better solution based on request listeners instead of exceptions.

File: src/Glpi/Http/Listener/CheckGlpiConfigListener.php [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+        // Check if the DB is configured properly
+        if ($this->skip_db_checks) {
+            return;
+        }
+
+        if (!\is_file(GLPI_CONFIG_DIR . '/config_db.php')) {
+            Session::loadLanguage('', false);
+
+            $event->setResponse(new StreamedResponse(new DisplayGlpiMisconfiguredPage($this->projectDir), 500));
```
Comment (cedric-anne) [inline src/Glpi/Http/Listener/CheckGlpiConfigListener.php:107]: The event propagation should be stopped here. We do not want other request listeners to be executed.

```suggestion
            $event->setResponse(new StreamedResponse($this->displayGlpiMisconfiguredPage(...), 500));
            $event->stopPropagation();
```

File: src/Glpi/Http/ListenersPriority.php [hunk 2]
```diff
@@ -34,11 +34,23 @@
...
+use Glpi\Http\Listener\RedirectLegacyRouteListener;
+use Glpi\Http\Listener\RootDocListener;
+use Glpi\Http\Listener\SessionStartListener;
 
 final class ListenersPriority
 {
     public const LEGACY_LISTENERS_PRIORITIES = [
+        RootDocListener::class => 700,
+        SessionStartListener::class => 700,
+        CheckGlpiConfigListener::class => 600,
```
Comment (cedric-anne) [inline src/Glpi/Http/ListenersPriority.php:52]: These 3 listeners should be moved after the `RedirectLegacyRouteListener` listener. Indeed, all previous listeners can be executed first, they do not require the DB to be installed/up-to-date.

File: src/Glpi/Http/Listener/CheckGlpiConfigListener.php (hunk 1 above)
Comment (Pierstoval) [reply src/Glpi/Http/Listener/CheckGlpiConfigListener.php:107]: Propagation is always stopped when a Response is added:

image


Also, the object is callable, and I might reuse it, so using a different callable isn't the best IMO. The goal of the callable object is to be able to be used elsewhere, and I'm not sure whether it's gonna be needed yet. If it's never gonna be used outside, I'll bring it back in the listener''s code

**Review by cedric-anne — APPROVED** [review]

**Review by AdrienClairembault — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]
I squashed commits and rebased on the `main` branch to simplify the developments that depend on it until this PR is merged.

**Review by trasher — APPROVED** [review]

**Pull request merged**